web/docs/deployment/local-testing.md

```diff
@@ -0,0 +1,136 @@
+| Client environment                       | Static server       | Static server                                     |
+| Assets                                   | Served individually | Bundled and minified                              |
+| React dev mode                           | Enabled             | Disabled                                          |
+| Hot reload                               | Enabled             | Disabled                                          |
+| Source maps                              | Enabled             | Disabled                                          |
+| Debugging support                        | Full                | Limited                                           |
+| Performance                              | Slower              | Normal                                            |
+
+## Passing environment variables
+
+You have to manually specify any environment variables that your app needs to run in production. This is crucial because the production build may require different configurations than the development build. This helps you make sure of which environment variables you need to set in your deployment provider for your app to work correctly.
+
+This includes database URLs, API keys, and any other configuration settings that are necessary for your app to function correctly. You can usually check out your [`.env` files](../project/env-vars.md#dotenv-files) to see what environment variables your app expects. You can read more about environment variables in Wasp in the [environment variables guide](../project/env-vars.md).
+
+The only exception are the environment variables that configure your app's client and server URLs (`WASP_WEB_CLIENT_URL`, `WASP_SERVER_URL`, and `REACT_APP_API_URL`). Because `wasp build start` knows that it's running the app on your local workstation, it can fill them out for you automatically.
+
+:::info Which values should I use when testing?
+
+- Do not use real production secrets or endpoints.
+- Prefer staging/sandbox credentials and services that mirror production (e.g., Stripe test keys, a staging DB or an isolated local DB with realistic data).
+- Keep config parity with production: same feature flags, callbacks/redirect URLs, and optional vars set/unset as in prod.
+
+Example:
+
+```bash
+wasp build start --server-env-file .env.staging --client-env-file .env.client.staging
+```
+
+:::
+
+### Server environment variables
+
+Use `--server-env` to specify environment variables for the server:
+
+```bash
+wasp build start --server-env DATABASE_URL=postgresql://localhost:5432/myapp
+```
+
+You can specify multiple server environment variables:
+
+```bash
+wasp build start --server-env DATABASE_URL=postgresql://localhost:5432/myapp --server-env JWT_SECRET=my-secret-key
+```
+
+You can also point to an `.env` file to load environment variables:
+
+```bash
+wasp build start --server-env-file .env.production
```

Comment:
Wow, nice! How do we do this, can it help us here: https://github.com/wasp-lang/wasp/issues/3026

Reply:
It's on the Haskell side not the JS side. We already had a `parseDotEnv` function, I just hooked it up to the CLI argument parser:

https://github.com/wasp-lang/wasp/blob/0c6b0be9b0f6b9b47e12595b6c4146e454bc2293/waspc/cli/src/Wasp/Cli/Util/EnvVarArgument.hs#L46-L47